### PR TITLE
Use drupal client for getting data

### PR DESCRIPTION
**Pull request by harumijang**

### CODE REVIEW

**Comment by harumijang** [comment]
Would it be good to have our version of the `get-params` function instead of importing DrupalJsonApiParams in every file with a request or is that not necessary right now

https://github.com/chapter-three/next-drupal/blob/main/examples/example-umami/lib/get-params.ts

**Comment by lauriii** [comment]
> Would it be good to have our version of the `get-params` function like how it is used in [https://github.com/chapter-three/next-drupal/blob/main/examples/example-umami/pages/articles/index.tsx#L54](https://github.com/chapter-three/next-drupal/blob/main/examples/example-umami/pages/articles/index.tsx?rgh-link-date=2022-06-28T22%3A55%3A29Z#L54) or is that not necessary right now

I think we could consider that in a follow-up but we shouldn't do it at least in the scope of this PR.

File: starters/basic-starter/lib/drupal.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { DrupalClient } from 'next-drupal';
+
+export const drupal = new DrupalClient(
+  process.env.NEXT_PUBLIC_DRUPAL_BASE_URL,
+  {
+    previewSecret: process.env.DRUPAL_PREVIEW_SECRET,
```
Comment (lauriii) [inline starters/basic-starter/lib/drupal.ts:6]: I'm getting following error when using the preview mode:

```
$ yarn start
yarn run v1.22.17
$ next start
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from /Users/lauri.eskola/Projects/next-acquia-cms/starters/basic-starter/.env.local
Error: auth is not configured. See https://next-drupal.org/docs/client/auth
    at /Users/lauri.eskola/Projects/next-acquia-cms/node_modules/next-drupal/dist/index.js:1177:23
    at /Users/lauri.eskola/Projects/next-acquia-cms/node_modules/next-drupal/dist/index.js:1285:12
    at DrupalClient.<anonymous> (/Users/lauri.eskola/Projects/next-acquia-cms/node_modules/next-drupal/dist/index.js:1287:8)
    at DrupalClient.fetch (/Users/lauri.eskola/Projects/next-acquia-cms/node_modules/next-drupal/dist/index.js:1121:21)
    at DrupalClient.translatePath (/Users/lauri.eskola/Projects/next-acquia-cms/node_modules/next-drupal/dist/index.js:1927:38)
    at DrupalClient.translatePathFromContext (/Users/lauri.eskola/Projects/next-acquia-cms/node_modules/next-drupal/dist/index.js:1950:38)
    at getStaticProps (/Users/lauri.eskola/Projects/next-acquia-cms/starters/basic-starter/.next/server/pages/[...slug].js:136:107)
    at Object.renderToHTML (/Users/lauri.eskola/Projects/next-acquia-cms/node_modules/next/dist/server/render.js:492:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async doRender (/Users/lauri.eskola/Projects/next-acquia-cms/node_modules/next/dist/server/base-server.js:901:38)
```

I think we still need to configurate the authentication here using instructions from https://next-drupal.org/docs/authentication

**Review by lauriii — CHANGES_REQUESTED** [review]

**Comment by lauriii** [comment]
We should also remove `DRUPAL_SITE_ID`, `DRUPAL_FRONT_PAGE`, and `DRUPAL_JSONAPI_PREFIX=/jsonapi` from the .env.example since they are no longer used after we start using the `DrupalClient`.

**Comment by lauriii** [comment]
Opened upstream issue for this too https://github.com/chapter-three/next-drupal/issues/247

**Comment by harumijang** [comment]
Looks good to me

**Review by lauriii — APPROVED** [review]

**Pull request merged**